File: src/Python/401-500/448.FindDisappearedNumbers.py

class Solution:
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        return list(set(range(1, len(nums) + 1)) - set(nums))
        
if __name__ == '__main__':
    solution = Solution()
    print(solution.findDisappearedNumbers([4,3,2,7,8,2,3,1]))
else:
    pass

File: src/CSharp/401-500/448.FindDisappearedNumbers.cs
using System;
using System.Collections.Generic;

namespace PeefyLeetCode.FindDisappearedNumbers
{

public class Solution {
    public IList<int> FindDisappearedNumbers(int[] nums) {
        var res = new List<int>();
        for (int i = 0; i < nums.Length; ++i) {
            while (nums[i] != (i+1) && nums[i] != nums[nums[i] - 1])
		    {
		        //	swap(nums[i], nums[nums[i] - 1]);
                int tmp = nums[i];
			    nums[i] = nums[nums[i] - 1];
			    nums[tmp - 1] = tmp;
		    }
        }
        for (int i = 0; i < nums.Length; ++i) {
            if (nums[i] != i + 1) {
                res.Add(i + 1);
            }
        }
        return res;
    }
}

}